pkg/csi_driver/gcfs_driver.go

```diff
@@ -133,13 +169,19 @@ func NewGCFSDriver(config *GCFSDriverConfig) (*GCFSDriver, error) {
 			csi.ControllerServiceCapability_RPC_CREATE_DELETE_SNAPSHOT,
 		}
 		driver.addControllerServiceCapabilities(csc)
+
+		if config.FeatureOptions.FeatureStateful != nil && config.FeatureOptions.FeatureStateful.Enabled {
+			recon, factory, coreFactory, driverFactory := initMultishareReconciler(config)
+			runMultishareReconciler(config, recon, factory, coreFactory, driverFactory)
```

Comment:
The split of reconciler struct and run looks good. But the original intention of the comment was to move the runMultishareReconciler() out of the NewGCFSDriver() function.

I think its fine to store the factor as non exported variables in the GCFSDriverConfig struct, which can then be re-used in GCFSDriver.Run() . Thoughts?

Reply:
make sense. doen